src/encoded/static/components/filegallery.js

```diff
@@ -2452,6 +2443,52 @@ class FileGalleryRendererComponent extends React.Component {
         return [];
     }
 
+    /**
+     * Sets the `selectedAnalysisProps` property to track the corresponding selected assembly/annotation
+     * and pipeline lab. Only pass in `compiledAnalyses` if it has been newly determined and might
+     * not yet have propogated to `this.state.compiledAnalyses`.
+     * @param {number} selectedAnalysesIndex New index of selected `compiledAnalyses`
+     * @param {array} compiledAnalyses Compiled analyses for the experiment if newly determined
+     */
+    saveSelectedAnalysesProps(selectedAnalysesIndex, compiledAnalyses) {
+        const localCompiledAnalyses = compiledAnalyses || this.state.compiledAnalyses;
+        const selectedAnalyses = localCompiledAnalyses[selectedAnalysesIndex];
+        this.selectedAnalysisProps = { assembly: selectedAnalyses.assembly, pipelineLab: selectedAnalyses.pipelineLab };
+    }
+
+    /**
+     * Find the compiled analyses object best matching the given assembly/annotation and the last
+     * selected pipeline lab and return its index into the compiled analyses array. Only pass in
+     * `compiledAnalyses` if it has been newly determined and might not yet have propogated to
+     * `this.state.compiledAnalyses`. If no appropriate compiled analyses can be found, return 0
+     * which selects the first compiled analyses object.
+     * @param {string} assembly Suggested assembly for matching an analysis
+     * @param {array} compiledAnalyses Compiled analyses for the experiment if newly determined
+     *
+     * @return {object} Array index of most appropriate compiled analyses object
+     */
+    findCompiledAnalysesIndex(assembly, compiledAnalyses) {
+        let compiledAnalysisIndex = 0;
+        const localCompiledAnalyses = compiledAnalyses || this.state.compiledAnalyses;
+
+        // Find all analyses matching the given assembly, then of those find one that also matches
+        // the pipeline lab.
+        const analysesWithMatchingAssembly = localCompiledAnalyses.filter(analysis => analysis.assembly === assembly);
+        if (analysesWithMatchingAssembly.length > 0) {
+            const matchingAnalysis = analysesWithMatchingAssembly.find(analysis => analysis.pipelineLab === this.selectedAnalysisProps.pipelineLab);
```

Comment:
Just curious, could there be multiple analyses with a matching assembly and pipeline lab?

Reply:
Not at this level because any entries with duplicate assembly and pipeline labs get their files combined into one entry. So one or the other must be different for their to be separate entries.